### PR TITLE
(baby-project) add docker dev envierment and docs for local env

### PR DESCRIPTION
**Pull request by dusan-turajlic**

### CODE REVIEW

**Comment by dusan-turajlic** [comment]
Just realized I should probably create a JPDA (Java Platform Debugger Architecture) mechanism for the `./gradlew bootRun`

File: README.md [hunk 1]
```diff
@@ -1,4 +1,21 @@
 # Getting Started
+Our dev environment can be used in two ways
+### Docker:
+```shell
+docker-compise up -d
```
Comment (teroyks) [inline README.md]: Just a typo 🙂
```suggestion
docker-compose up -d
```

File: src/test/kotlin/turku/forge/babyproject/AppTests.kt [hunk 2]
```diff
@@ -1,15 +1,13 @@
-package turku.forge.babyProject
+package turku.forge.babyproject
```
Comment (teroyks) [inline src/test/kotlin/turku/forge/babyproject/AppTests.kt:1]: **praise:** lower case package names are better

**Comment by teroyks** [comment]
Looks pretty good!

Not many comments in the code though 🙂 – even though this looks pretty standard, I would appreciate us adopting the practice of including a short description for each class – just their purpose is enough for simple ones; still makes browsing the code nicer (IMO)

File: README.md [hunk 3]
```diff
@@ -1,4 +1,21 @@
 # Getting Started
+Our dev environment can be used in two ways
+### Docker:
```
Comment (teroyks) [inline README.md]: **nitpick:** Would prefer semantically correct subheadings

A 1st level heading should be followed by a 2nd level subheading.

**Comment by teroyks** [comment]
**thought (non-blocking):** Next steps

I like that there are already instructions on building the thing. My wishlist for next steps (outside this PR) would be running tests and how to access the server.

**Comment by teroyks** [comment]
**question:** Is the server supposed to run already?

I don’t know if I’m getting ahead of myself, but on startup, I got the following error:
```log
Attaching to baby-project-server_backend_1
backend_1  | Compilation with Kotlin compile daemon was not successful
backend_1  | java.rmi.ServerError: Error occurred in server thread; nested exception is:
backend_1  | 	java.lang.AssertionError: rootDir must be verified to be directory beforehand.
```

**Comment by dusan-turajlic** [comment]
> question: Is the server supposed to run already?

Yes, it is, that looks like a Gradle error, could be that my local env has something the helps the docker run, I'll install the project on a different environment and see what might be missing. Could be something trivial like access to /app in docker

**Comment by dusan-turajlic** [comment]
> **thought (non-blocking):** Next steps
> 
> 
> 
> I like that there are already instructions on building the thing. My wishlist for next steps (outside this PR) would be running tests and how to access the server.

Ahh, yea, forgot about how to run the tests will work in the docker env. I need to google the correct way to do this 😅

You can in the startup.sh remove the `-x test` and it will run the test after you build and before it’s deployed and that happens on save. That's great for TDD but I figured we don’t want to force people to do TDD just recommend it, we need go put that behind some evierment variable at some point.

I can create an issue for that so it doesn't get forgotten

**Review by teroyks — APPROVED** [review]

**Pull request merged**